### ISSUE TITLE
/opt/homebrew/bin/brew shellenv sets $MANPATH which hides Mac OS shipped manages

### ISSUE THREAD

**Issue by dwt** [Homebrew/brew#13235]
### `brew config` output

```shell
HOMEBREW_VERSION: 3.4.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3088bbf1a696961cd4202d9ef85ac5f95b977e7c
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5368053deffbb6e923ebbc907bad6bc98d9b1e2a
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: mate -w
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3.1-arm64
CLT: 13.0.0.0.1.1627064638
Xcode: 13.3.1
Rosetta 2: false
```

### `brew doctor` output

```shell
Your system is ready to brew.
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I want my system manpages visible as well as the Homebrew ones.

### What happened (include all command output)?

After a fresh install of Homebrew, it instructs to activate the Homebrew environment by adding

```
eval "$(/opt/homebrew/bin/brew shellenv)"
```

into the profile.

This is problematic because this contains the following line:

```
% env -i /opt/homebrew/bin/brew shellenv |grep MANPATH
export MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:";
```

Setting this variable (which is by default unset on macOS) effectively hides all man pages that the system provides, as the the system then ignores the options set in `/private/etc/man.conf` which derive the man path of the system from the `$PATH` variable.

For example: `man rsync` fails after installation of Homebrew, but works again after executing `unset MANPATH`.

### What did you expect to happen?

`man rsync` (or indeed any system provided manpages) should continue to work just fine after installation of Homebrew.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
- install Homebrew according to brew.sh, i.e. `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"` on a fresh install macOS Monterey 12.3.1 (21E258)
- `man rsync` or `man ls` or any other MacOS provided man page
```

**Comment by dwt** [comment]
With limited testing it seems that after `unset MANPATH` the Homebrew provided manpages are still visible with man. The system by which man derives the manpath from `$PATH` may therefore be good enough to cover what Homebrew needs.

**Comment by dwt** [comment]
This is also described in the section `SEARCH PATH FOR MANUAL PAGES` in `man man`

**Comment by carlocab** [comment]
Can't reproduce.
```
$ echo $MANPATH
/opt/homebrew/share/man::
$ which -a rsync
/usr/bin/rsync
```
`man rsync` works just fine for me.

**Comment by Bo98** [comment]
> Setting this variable (which is by default unset on macOS) effectively hides all man pages that the system provides, as the the system then ignores the options set in `/private/etc/man.conf` which derive the man path of the system from the `$PATH` variable.

The colon at the end of `MANPATH` should indicate that the defaults are to be appended to the end.

```
MANPATH="/opt/homebrew/share/man${MANPATH+:$MANPATH}:" man -dw rsync
Reading config file /private/etc/man.conf
Looked whether there exists a message catalog man, but there is none
(and for English messages none is needed)

found man directory /usr/share/man
found man directory /usr/local/share/man
found man directory /usr/X11/man
found manpath map /bin --> /usr/share/man
found manpath map /sbin --> /usr/share/man
found manpath map /usr/bin --> /usr/share/man
found manpath map /usr/sbin --> /usr/share/man
found manpath map /usr/local/bin --> /usr/local/share/man
found manpath map /usr/local/sbin --> /usr/local/share/man
found manpath map /usr/X11/bin --> /usr/X11/man
found manpath map /usr/bin/X11 --> /usr/X11/man
found manpath map /usr/bin/mh --> /usr/share/man

using /usr/bin/less -is as pager

using /usr/bin/less -is as browser

using /bin/cat to dump HTML pages as textadding /opt/homebrew/share/man to manpath

path directory /usr/local/bin is in the config file
adding /usr/local/share/man to manpath

path directory /usr/bin is in the config file
adding /usr/share/man to manpath

path directory /bin is in the config file

path directory /usr/sbin is in the config file

path directory /sbin is in the config file

path directory /Applications/VMware Fusion.app/Contents/Public is not in the config file
and we found no man directory nearby

path directory /Library/TeX/texbin is not in the config file
but there is a man directory nearby
adding /Library/TeX/texbin/man to manpath

path directory /opt/X11/bin is not in the config file
but there is a man directory nearby
adding /opt/X11/share/man to manpath

adding mandatory man directories

adding /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/share/man to manpath
adding /Applications/Xcode.app/Contents/Developer/usr/share/man to manpath
adding /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man to manpath
will try to write /var/cache/man/cat1/rsync.1.bz2 if needed
status from is_newer() = -2
/usr/share/man/man1/rsync.1.gz
```


```
MANPATH="/usr/local/opt/ruby/share/man" man -dw ruby
Reading config file /private/etc/man.conf
Looked whether there exists a message catalog man, but there is none
(and for English messages none is needed)

found man directory /usr/share/man
found man directory /usr/local/share/man
found man directory /usr/X11/man
found manpath map /bin --> /usr/share/man
found manpath map /sbin --> /usr/share/man
found manpath map /usr/bin --> /usr/share/man
found manpath map /usr/sbin --> /usr/share/man
found manpath map /usr/local/bin --> /usr/local/share/man
found manpath map /usr/local/sbin --> /usr/local/share/man
found manpath map /usr/X11/bin --> /usr/X11/man
found manpath map /usr/bin/X11 --> /usr/X11/man
found manpath map /usr/bin/mh --> /usr/share/man

using /usr/bin/less -is as pager

using /usr/bin/less -is as browser

using /bin/cat to dump HTML pages as textadding /usr/local/opt/ruby/share/man to manpath
will try to write /var/cache/man/local/opt/ruby/cat1/ruby.1.bz2 if needed
status from is_newer() = -2
/usr/local/opt/ruby/share/man/man1/ruby.1
```

**Comment by dwt** [comment]
Thanks @Bo98 for the analysis! Due to that I found out that the problem is actually a special case of the behavior of ZSH, where the `ENVIRONMENT_VARIABLES` variables are also mapped to `environment_variables` that can be used with array syntax. Using those removes the empty path element at the end of the `MANPATH`. :-(

In my case I had this later in the config: `manpath=(~/bin/man $manpath)` which I replaced with `export MANPATH="~/bin/man:$MANPATH"`.

Now everything works again.

So... is this a bug in ZSH that it swallows empty array entries erroneously?
